it/src/main/java/org/corfudb/universe/util/DockerManager.java

```diff
@@ -134,8 +134,8 @@ public void restart(String containerName) {
 
         try {
             ContainerInfo container = docker.inspectContainer(containerName);
-            if (container.state().running() || container.state().paused()) {
-                log.warn("The container `{}` already running, should stop before restart", container.name());
+            if (container.state().restarting()) {
```

Comment:
Logically it makes sense to be able to restart a running container. The old one however was preventing the restart of running or paused containers. Was that by design or a bug?

Reply:
It's a bug. We have both `start()` and `restart()`. `start()` is for stopped containers, while `restart()` is for running ones. But the old logic is same for these two, so it should be a bug.